docs/design/unified_sidecar_and_process_monitor.md

```diff
@@ -0,0 +1,118 @@
+
+When we are transitioning from the old launcher to the new launcher for a cluster, we will include both a monitor conf file and the kubernetes-monitor conf file in the config map. Once the transition is complete, we will remove the monitor conf file from the sidecar.
+
+### Process Configuration
+
+The process configuration will be represented in a JSON file that contains a ProcessConfiguration object. The spec for this file is describe below.
+
+	ProcessConfiguration {
+		version: string - The version of FoundationDB the process should run. This will determine the path to the fdbserver process.
+		arguments: []Argument - The arguments to the fdbserver process
+	}
+
+	Argument {
+		type: ArgumentType - Determines how the value is generated. The default is Literal
+		value: string - For Literal arguments, the value for the argument
+		values: []Argument - For Concatenate arguments, a list of smaller arguments that are combined to form a single argument
+		source: string - For Environment arguments, the name of an environment variable to use as the source of the argument.
+		multiplier: integer - For ProcessNumber arguments, a multiplier to apply to the process number
+		offset: integer - For ProcessNumber arguments, an offset to apply to the process number
+	}
+
+	ArgumentType enum {
+		Literal - A value provided in the argument
+		Concatenante - A combination of other arguments
+		Environment - A value from an environment variable
+		ProcessNumber - A value calculated from the process number. The processes are numbered starting from 1.
+	}
+
+Example configuration, in JSON form:
+
+	{
+		"version": "6.3.0",
+		"arguments": [
+			{"value": "--public-address"},
+			{"type": "Concatenate", "values": [
+				{"type": "EnvVar", "source": "FDB_PUBLIC_IP"},
+				{"value": ":"},
+				{"type": "ProcessNumber", "offset": 4498, "multiplier": 2},
+				{"value": ":tls"}
+			]},
+			{"value": "--listen-address"},
+			{"type": "Concatenate", "values": [
+				{"type": "EnvVar", "source": "FDB_POD_IP"},
+				{"value": ":"},
+				{"type": "ProcessNumber", "offset": 4498, "multiplier": 2},
+				{"value": ":tls"}
+			]},
+			{"value": "--datadir"},
+			{"type": "Concatenate", "values": [
+				{"value": "/var/fdb/data/"},
+				{"type": "ProcessNumber"}
+			]},
+			{"value": "--class"},
+			{"value": "storage"},
+			{"value": "--locality-zoneid"},
+			{"type": "Environment", "source": "FDB_ZONE_ID"},
+			{"value": "--locality-instance-id"},
+			{"type": "Environment", "source": "FDB_INSTANCE_ID"},
+			{"value": "--locality-process-id"},
+			{"type": "Concatenate", "values": [
+				{"type": "Environment", "source": "FDB_INSTANCE_ID"},
+				{"value": "-"},
+				{"type": "ProcessNumber"}
+			},
+		]
+	}
+
+Let's assume this is run with a process count of 2, that 6.3.0 is the version of the main container, and that the following environment variables are set:
+
+	FDB_PUBLIC_IP=127.0.0.1
```

Comment:
I would rather use here `10.0.0.1` and `192.168.0.1` as Pod IP to make it more obvious.